### ISSUE TITLE
Multi-index search

### ISSUE THREAD

**Issue by gl0gl0** [algolia/react-instantsearch#55]
Hi,

When implementing multi-index search, following the example:
https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch/examples/multi-index/src/App.js

My code:
https://screencast.com/t/emslqzaPHH

The hit results are showing correctly for the second index (artists), but for the first index (songs) no results are being displayed. And I see that in the Network tab (Developer Tools), the REST calls shows hit results for both of them.

Any help would be very much appreciated.
Thanks!

**Comment by mthuret** [comment]
Thanks for spotting this regression @gl0gl0! 

Right now, It will work with this syntax: 

```js
<InstantSearch
      appId=""
      apiKey=""
      indexName="index1"
    >
      <SearchBox />
      <Index indexName="index1">
        <CustomHits />
      </Index>
      <Index indexName="index2">
        <CustomHits />
      </Index>
    </InstantSearch>
```

But definitely the syntax you used should work! I will fix it, and it will be part of a new beta.

**Comment by mthuret** [comment]
Should be fix with 4.0.0-beta.4
